dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectService.java

```diff
@@ -36,57 +36,31 @@
                                                                                                                                                         "service.ranking:Integer=9999" })
 public class CDIManagedObjectService implements ManagedObjectService {
 
-    private final AtomicServiceReference<CDIService> cdiServiceRef = new AtomicServiceReference<CDIService>("cdiService");
-    private final AtomicServiceReference<DefaultManagedObjectService> defaultMOSRef = new AtomicServiceReference<DefaultManagedObjectService>("defaultManagedObjectService");
+    private DefaultManagedObjectService defaultMOS;
+    private CDIService cdiService;
 
     private CDIRuntime cdiRuntime;
```

Comment:
Make `defaultMOS` and `cdiRuntime` final, remove `cdiService`  and add an `@Activate` constructor that sets these final values with constructor injection:

```java
@Activate
public CDIManagedObjectService(@Reference CDIService cdiService,
                               @Reference DefaultManagedObjectService defaultMOS) {
    this.cdiRuntime =  (CDIRuntime) cdiService;
    this.defaultMOS = defaultMOS;
}
```

The default for reference is static and mandatory, no need to name the reference.  Keep it simple.